src/core/WakaTimeCore.ts

```diff
@@ -348,6 +349,14 @@ class WakaTimeCore {
     return payload;
   }
 
+  getOperatingSystem(): Promise<string> {
+    return new Promise((resolve) => {
```

Comment:
Please don't wrap code in Promise functions, Let's keep using async/await over it

Reply:
I did this so we could use `await` when calling `chrome.runtime.getPlatformInfo` to turn it into an `async` function.